src/sql/workbench/contrib/notebook/browser/cellViews/textCell.component.ts

```diff
@@ -551,12 +560,24 @@ function preventDefaultAndExecCommand(e: KeyboardEvent, commandId: string) {
 export class RichTextEditStack {
 	private _list: string[] = [];
 
+	constructor(private _maxStackSize: number) {
+	}
+
+	public set maxStackSize(stackSize: number) {
+		this._maxStackSize = stackSize;
+	}
+
 	/**
-	 * Adds an element to the top of the stack.
+	 * Adds an element to the top of the stack. If the number of elements
+	 * exceeds the max stack size, then the oldest elements are removed until
+	 * the max size is reached.
 	 * @param element The string element to add to the stack.
 	 */
 	public push(element: string): void {
 		this._list.push(element);
+		if (this._list.length > this._maxStackSize) {
```

Comment:
I feel like we should actually calculate the size of the string and have the "max size" be based off that instead. Because having 10 elements that are 5 characters long is a whole lot different than 10 elements that are 5 pages of text. 

Or maybe even have that as a separate setting, since most of the time this kind of thing won't come up (it would just suck when it did)

Reply:
If you're changing the stack size on the fly, then that could remove older elements from the history. Then the user might not be able to undo to a certain point and it wouldn't be obvious why. At least in this case the limit is constant and obvious to the user. VSCode doesn't have Undo history limits as far as I know, so this is more of an edge case perf compromise for us.

Reply:
Yeah they don't seem to have limits, but that's because what they store is limited in scope (it's just changes)

Anyways while I'm not sure that what you said is any "worse" than having the app eat up a tons of memory if you try to edit some extremely large cells (after all, the edge case in both scenarios is basically the same) this is at least an improvement for the majority of users so guess it's fine.